sbi/inference/trainers/marginal/marginal_base.py

```diff
@@ -0,0 +1,307 @@
+
+import torch
+from torch import Tensor
+from torch.nn.utils.clip_grad import clip_grad_norm_
+from torch.optim.adam import Adam
+from torch.utils import data
+from torch.utils.data.sampler import SubsetRandomSampler
+from torch.utils.tensorboard.writer import SummaryWriter
+
+from sbi.inference.trainers.base import NeuralInference
+from sbi.neural_nets.estimators import UnconditionalDensityEstimator
+from sbi.neural_nets.estimators.shape_handling import (
+    reshape_to_batch_event,
+)
+from sbi.neural_nets.net_builders.flow import build_zuko_unconditional_flow
+from sbi.sbi_types import Shape
+from sbi.utils import check_estimator_arg
+from sbi.utils.torchutils import assert_all_finite
+
+
+def marginal_nn(
+    model: str,
+    z_score_x: Optional[str] = "independent",
+    hidden_features: int = 50,
+    num_transforms: int = 5,
+    num_bins: int = 10,
+    num_components: int = 10,
+    **kwargs: Any,
+) -> Callable:
+    r"""
+    Returns a function that builds a density estimator for learning the marginal.
+
+    Args:
+        model: The type of density estimator that will be created. One of [`mdn`,
+            `made`, `maf`, `maf_rqs`, `nsf`].
+        z_score_x: Whether to z-score samples $x$ before passing them into
+            the network, can take one of the following:
+            - `none`, or None: do not z-score.
+            - `independent`: z-score each dimension independently.
+            - `structured`: treat dimensions as related, therefore compute mean and std
+            over the entire batch, instead of per-dimension. Should be used when each
+            sample is, for example, a time series or an image.
+        hidden_features: Number of hidden features.
+        num_transforms: Number of transforms when a flow is used. Only relevant if
+            density estimator is a normalizing flow (i.e. currently either a `maf` or a
+            `nsf`). Ignored if density estimator is a `mdn` or `made`.
+        num_bins: Number of bins used for the splines in `nsf`. Ignored if density
+            estimator not `nsf`.
+        num_components: Number of mixture components for a mixture of Gaussians.
+            Ignored if density estimator is not an mdn.
+        kwargs: additional custom arguments passed to downstream build functions.
+    """
+
+    kwargs = dict(
+        zip(
+            (
+                "z_score_x",
+                "hidden_features",
+                "num_transforms",
+                "num_bins",
+                "num_components",
+            ),
+            (
+                z_score_x,
+                hidden_features,
+                num_transforms,
+                num_bins,
+                num_components,
+            ),
+            strict=False,
+        ),
+        **kwargs,
+    )
+
+    def build_fn(batch_x):
+        return build_zuko_unconditional_flow(which_nf=model, batch_x=batch_x)
+
+    return build_fn
+
+
+class MarginalEstimator(NeuralInference, ABC):
```

Comment:
seeing this now and realizing that it is so fundamentally different from `NeuralInference`, i.e., most importantly, it does not have `theta` and `x`, but just `x` (could be `theta` as well), I think it should not inherit from `Neuralnference`, but be a standalone class for now. 

Then, because it is a `Estimator` in essense, it should live in `neural_nets/estimators` next to the `ConditionalEstimator`.